### PR TITLE
docs(oidc): add freshrss oidc integration guide

### PR DESCRIPTION
**Pull request by darkpixelftw**
Create integration guide for oidc authentication for freshrss and authelia

Document non-obvious behaviour that freshrss expects port in the redirect URI

Signed-off-by: darkpixelftw <darkpixelftw@outlook.com>

### CODE REVIEW

**Comment by james-d-elliott** [comment]
Really appreciate the contribution! If you have time I'd appreciate confirming the token endpoint auth method this client uses. We will likely be requiring this to be configured in authelia in the next release.

**Comment by darkpixelftw** [comment]
Hi, I asked the maintainer of FreshRSS what the token auth method is and it seems we're having a bit of trouble answering this, see: https://github.com/FreshRSS/FreshRSS/issues/6102

**Comment by james-d-elliott** [comment]
Thanks, it's `client_secret_basic`. I'll make the recommendations based on this.

File: docs/content/en/integration/openid-connect/freshrss/index.md [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    parent: "openid-connect"
+weight: 620
+toc: true
+community: true
+---
+
+## Tested Versions
+
+* [Authelia]
+  * [v4.37.5](https://github.com/authelia/authelia/releases/tag/v4.37.5)
```
Comment (james-d-elliott) [inline docs/content/en/integration/openid-connect/freshrss/index.md:19]:
```suggestion
  * [v4.38.0](https://github.com/authelia/authelia/releases/tag/v4.38.0)
```

File: docs/content/en/integration/openid-connect/freshrss/index.md [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+      public: false
+      authorization_policy: two_factor
+      redirect_uris:
+        - https://freshrss.example.com:443/i/oidc/ # IMPORTANT: you must add the port to the redirect uri
+      scopes:
+        - openid
+        - groups
+        - email
+        - profile
+      userinfo_signed_response_alg: 'none'
```
Comment (james-d-elliott) [inline docs/content/en/integration/openid-connect/freshrss/index.md:64]:
```suggestion
      token_endpoint_auth_method: 'client_secret_post'
      userinfo_signed_response_alg: 'none'
```

File: docs/content/en/integration/openid-connect/freshrss/index.md [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+      secret: '$pbkdf2-sha512$310000$c8p78n7pUMln0jzvd4aK4Q$JNRBzwAo0ek5qKn50cFzzvE9RXV88h1wJn5KGiHrD0YKtZaR/nCb2CJPOsKaPK0hjf.9yHxzQGZziziccp6Yng'  # The digest of 'insecure_secret'.
+      public: false
+      authorization_policy: two_factor
+      redirect_uris:
+        - https://freshrss.example.com:443/i/oidc/ # IMPORTANT: you must add the port to the redirect uri
+      scopes:
+        - openid
+        - groups
+        - email
+        - profile
```
Comment (james-d-elliott) [inline docs/content/en/integration/openid-connect/freshrss/index.md:63]:
```suggestion
        - 'openid'
        - 'groups'
        - 'email'
        - 'profile'
```

File: docs/content/en/integration/openid-connect/freshrss/index.md [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+    ## The other portions of the mandatory OpenID Connect 1.0 configuration go here.
+    ## See: https://www.authelia.com/c/oidc
+    clients:
+    - id: freshrss
+      description: freshrss
+      secret: '$pbkdf2-sha512$310000$c8p78n7pUMln0jzvd4aK4Q$JNRBzwAo0ek5qKn50cFzzvE9RXV88h1wJn5KGiHrD0YKtZaR/nCb2CJPOsKaPK0hjf.9yHxzQGZziziccp6Yng'  # The digest of 'insecure_secret'.
+      public: false
+      authorization_policy: two_factor
+      redirect_uris:
+        - https://freshrss.example.com:443/i/oidc/ # IMPORTANT: you must add the port to the redirect uri
```
Comment (james-d-elliott) [inline docs/content/en/integration/openid-connect/freshrss/index.md:58]:
```suggestion
        - 'https://freshrss.example.com:443/i/oidc/'
```

File: docs/content/en/integration/openid-connect/freshrss/index.md [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+identity_providers:
+  oidc:
+    ## The other portions of the mandatory OpenID Connect 1.0 configuration go here.
+    ## See: https://www.authelia.com/c/oidc
+    clients:
+    - id: freshrss
+      description: freshrss
+      secret: '$pbkdf2-sha512$310000$c8p78n7pUMln0jzvd4aK4Q$JNRBzwAo0ek5qKn50cFzzvE9RXV88h1wJn5KGiHrD0YKtZaR/nCb2CJPOsKaPK0hjf.9yHxzQGZziziccp6Yng'  # The digest of 'insecure_secret'.
+      public: false
+      authorization_policy: two_factor
```
Comment (james-d-elliott) [inline docs/content/en/integration/openid-connect/freshrss/index.md:56]:
```suggestion
      authorization_policy: 'two_factor'
```

File: docs/content/en/integration/openid-connect/freshrss/index.md [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+will operate with the above example:
+
+```yaml
+identity_providers:
+  oidc:
+    ## The other portions of the mandatory OpenID Connect 1.0 configuration go here.
+    ## See: https://www.authelia.com/c/oidc
+    clients:
+    - id: freshrss
+      description: freshrss
```
Comment (james-d-elliott) [inline docs/content/en/integration/openid-connect/freshrss/index.md:53]:
```suggestion
    - id: 'freshrss'
      description: 'freshrss'
```

File: docs/content/en/integration/openid-connect/freshrss/index.md [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+
+This example makes the following assumptions:
+
+* __Application Root URL:__ `https://freshrss.example.com/`
+* __Authelia Root URL:__ `https://auth.example.com/`
+* __Client ID:__ `freshrss`
+* __Client Secret:__ `insecure_secret`
+* __Port:__ '443'
+  * this is the port freshrss is served over (usually 80 for http and 443 for https) NOT the port of the docker container
+
```
Comment (james-d-elliott) [inline docs/content/en/integration/openid-connect/freshrss/index.md:37]:
```suggestion

### Special Notes

1. The FressRSS implementation seems to always include the port in the requested `redirect_uri`. As Authelia strictly conforms to the specifications this means the client registration **_MUST_** include the port for the requested `redirect_uri` to match.

```

**Review by james-d-elliott — CHANGES_REQUESTED** [review]
Thanks so much for contributing! Just a few changes. I know you've not tested with v4.38.0 but we're only accepting PR's with this version in mind as we're pretty close to a release.

If you'd like to verify this I can probably help you. Otherwise we can wait for the release and you can reevaluate.

**Comment by darkpixelftw** [comment]
> Thanks so much for contributing! Just a few changes. I know you've not tested with v4.38.0 but we're only accepting PR's with this version in mind as we're pretty close to a release.
> 
> If you'd like to verify this I can probably help you. Otherwise we can wait for the release and you can reevaluate.

Sure it would be a great help if you could verify it on the 4.38 version, If not then I'll just wait for the release. Is there an ETA for when the next release is?

File: docs/content/en/integration/openid-connect/freshrss/index.md [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+      public: false
+      authorization_policy: 'two_factor'
+      redirect_uris:
+        - 'https://freshrss.example.com:443/i/oidc/'
+      scopes:
+        - 'openid'
+        - 'groups'
+        - 'email'
+        - 'profile'
+      token_endpoint_auth_method: 'client_secret_post'
```
Comment (james-d-elliott) [inline docs/content/en/integration/openid-connect/freshrss/index.md:68]:
```suggestion
      token_endpoint_auth_method: 'client_secret_basic'
```

**Review by james-d-elliott — CHANGES_REQUESTED** [review]

**Comment by james-d-elliott** [comment]
We don't have a specific ETA, but it should be soon. We can try it against the beta but you'll need a database backup.

**Pull request merged**